op-supervisor/supervisor/backend/db/fromda/update.go

```diff
@@ -196,11 +238,35 @@ func (db *DB) addLink(source eth.BlockRef, derived eth.BlockRef, invalidated com
 		return err
 	}
 	if last.invalidated {
-		return fmt.Errorf("cannot build %s on top of invalidated entry %s: %w", link, last, types.ErrConflict)
+		return fmt.Errorf("cannot build %s on top of invalidated entry %s: %w", link, last, types.ErrAwaitReplacementBlock)
 	}
 	lastSource := last.source
 	lastDerived := last.derived
 
+	if revision.Any() {
+		link.revision = last.revision
+	} else {
+		if invalidated == (common.Hash{}) {
+			// Cross-safe db can bump the revision number without invalidating anything at the same time.
+			// But can only do so when inserting the expected entry
+			if last.revision != revision {
+				if derived.Number != revision.Number() {
+					return fmt.Errorf("cannot insert entry %s with revision %s, after %s with revision %s: %w",
+						derived, revision, last.derived, last.revision, types.ErrDataCorruption)
+				}
+				db.log.Warn("New revision", "revision", revision, "source", source, "derived", derived)
+			}
+		} else if derived.Hash == invalidated {
+			if last.revision == revision {
+				return fmt.Errorf("invalidated link %s should change revision: %w", link, types.ErrConflict)
```

Comment:
I'm a little unsure of ErrConflict here semantically, but functionally its likely the best match we have.